File: src/helper/Parser.java
package helper;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class Parser {

	public static int extractStatus(String in) {
		int ret = 0;

		try {
			ret = Integer.parseInt(extract(in, ".*status\":(\\d+),\".*"));
		} catch (NumberFormatException e) {
		}

		return ret;
	}

	public static Double extractConfidence(String in) {
		Double ret = 0.0;

		try {
			ret = Double
					.parseDouble(extract(in, ".*confidence\":(\\d.\\d+).*"));
		} catch (NumberFormatException e) {
		}

		return ret;
	}

	public static String extractUtterance(String in) {
		return extract(in, ".*utterance\":\"([a-zA-Z ]+)\",\"confidence.*");
	}

	public static String extract(String in, String regex) {
		String ret = "";
		Pattern p = Pattern.compile(regex);
		Matcher m = p.matcher(in);

		if (m.find()) {
			ret = m.group(1);
		}

		return ret;
	}
}

File: src/helper/Parser.scala
package helper

import java.util.regex.Pattern

/** Der Parser für die Antworten von Google auf die Audio-Aufnahmen */
object Parser {
  val ERR = -1

  /**
   * Parst den Status als Int aus einem String
   * @param in Der String aus dem der Wert extrahiert werden soll
   * @return Der extrahierte Wert als Int, -1 im Fehlerfall
   */
  def extractStatus(in: String) = {
    try {
      extract(in, ".*status\":(\\d+),\".*").toInt
    } catch {
      case e: Throwable => ERR
    }
  }

  /**
   * Parst die Zufriedenheit der Spracherkennunga als Int aus einem String
   * @param in Der String aus dem der Wert extrahiert werden soll
   * @return Der extrahierte Wert als Int, -1 im Fehlerfall
   */
  def extractConfidence(in: String) = {
    try {
      extract(in, ".*confidence\":(\\d.\\d+).*").toDouble
    } catch {
      case e: Throwable => ERR
    }
  }

  /**
   * Parst das Ergebnis der Spracherkennung aus einem String
   * @param in Der String aus dem der Text extrahiert werden soll
   * @return Der erkannte Text als String, ein leerer String im Fehlerfall
   */
  def extractUtterance(in: String) = {
    extract(in, ".*utterance\":\"(.*)\",\"confidence.*")
  }

  /**
   * Extrahiert mit Regex einen String aus einem anderen String
   * @param in Der String aus dem der Text extrahiert werden soll
   * @return Der erkannte Text als String, ein leerer String im Fehlerfall
   */
  def extract(in: String, regex: String): String = {

    var ret = "";
    val p = Pattern.compile(regex);
    val m = p.matcher(in);

    if (m.find()) {
      ret = m.group(1);
    }
    ret
  }
  
  /**
   * Beispielaufruf
   */
   def main(args: Array[String]) = {
    val in = "{\"status\":0,\"id\":\"d7cfe0647045cb1462f892e468fa3134-1\",\"hypotheses\":[{\"utterance\":\"rar rar rar rar rar rar rar\",\"confidence\":0.5146254}]}"
    println(in)

    println(extractStatus(in))
    println(extractConfidence(in))
    println(extractUtterance(in))
  }
}